### ISSUE TITLE
Add support for more complex shapes

### ISSUE THREAD

**Issue by DRSchlaubi** [Traewelling/line-colors#27]
https://de.wikipedia.org/wiki/U-Bahn_M%C3%BCnchen#Linie_U7
https://de.wikipedia.org/wiki/U-Bahn_M%C3%BCnchen#Linie_U8

**Comment by jheubuch** [comment]
🥲

**Comment by jheubuch** [comment]
So.. if you have an idea, feel free to contribute :D but I guess, these Munich lines are like an edge-case. would be nice if those could be displayed, but I actually dont know how..

**Comment by DRSchlaubi** [comment]
I enjoy suffering

**Comment by DRSchlaubi** [comment]
Found a way

**Comment by jheubuch** [comment]
Yeah, I get what this is doing, but I guess we need to find a way for non-Webapplications to render this correctly and convenient without having parsing issues...

**Comment by DRSchlaubi** [comment]
I don't feel like writing my own format for representing gradients

**Comment by tsia** [comment]
How about SVG? Could represent all the weird and wonderful shapes and colors that some designers dream up

**Comment by mrpelz** [comment]
SVG representations would basically treat these not as pictograms that are part of the displaying app’s design system but as icons that are ultimately defined by the operator (slightly different definitions for “pill” included).

But that’s fine, because their purpose is making what users encounter on screen resemble offline equivalents as closely as possible.

SVGs are also infinitely scalable and not limited to Web-based rendering.

**Comment by oneiricbotcelot** [comment]
I think another new shape which would represent a lot of bus lines is the regular hexagon. Examples can be found in [Stuttgarts bus map](https://download.vvs.de/SSB-Busliniennetz_A0.pdf).

**Comment by jheubuch** [comment]
I think hexagon already is on the specs (see readme down below) but I think nothing can render it atm (so please dont add it for the moment)

**Comment by oneiricbotcelot** [comment]
A hexagon is on the specs indeed, but not a **regular** hexagon. I won't add corresponding lines until the shape is officially supported.

**Comment by vainamov** [comment]
> I think hexagon already is on the specs (see readme down below) but I think nothing can render it atm (so please dont add it for the moment)

In my opinion we shouldn't disapprove adding data to the spec, just because it can't be rendered. Applications should ideally fallback to methods they support if they encounter unrecognized/non-implemented shapes.

Besides, there might be other use cases apart from rendering, that could benefit from this data nonetheless.
